R/dataframe.R

```diff
@@ -157,6 +160,9 @@ as_data_frame.data.frame <- function(x, ...) {
 #' @export
 #' @rdname as_data_frame
 as_data_frame.list <- function(x, validate = TRUE, ...) {
+
+  x <- compact(x)
```

Comment:
@hadley: NULL columns in data frames are difficult to create with base R, and they crash dplyr:

```
structure(list(a=NULL), class = "data.frame") %>% dplyr::bind_rows(data.frame(a=1))
```

While lst() handles NULL values just fine, I think NULL columns are not appropriate for data frames.